pkg/server/server.go

```diff
@@ -532,3 +541,79 @@ func goContext(parent genericapiserver.PostStartHookContext) context.Context {
 func (s *Server) WaitForPhase1Finished() {
 	<-s.rootPhase1FinishedCh
 }
+
+func (s *Server) leaderElectAndStartControllers(ctx context.Context) {
+	logger := klog.FromContext(ctx)
+
+	leaderElectionConfig := rest.CopyConfig(s.GenericConfig.LoopbackClientConfig)
+	leaderElectionConfig = rest.AddUserAgent(leaderElectionConfig, "kcp-leader-election")
+	// TODO(embik): is there a better way to access / generate the system:admin workspace url?
+	leaderElectionConfig.Host = fmt.Sprintf("%s/clusters/system:admin", leaderElectionConfig.Host)
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		logger.Error(err, "failed to determine hostname")
```

Comment:
any way to make this fatal, or at least bubble it up in the stack?

Reply:
I've moved this to before the goroutine and pass it as parameter now. The post-start-hook will bubble up this error now.